lib/services/local/headlessauthn_watcher.go

```diff
@@ -219,33 +223,34 @@ func (h *HeadlessAuthenticationWatcher) notify(headlessAuthns ...*types.Headless
 	}
 }
 
+type WaitConditionFunc func(*types.HeadlessAuthentication) (bool, error)
+
 // Wait watches for the headless authentication with the given id to be added/updated
 // in the backend, and waits for the given condition to be met, to result in an error,
 // or for the given context to close.
-func (h *HeadlessAuthenticationWatcher) Wait(ctx context.Context, name string, cond func(*types.HeadlessAuthentication) (bool, error)) (*types.HeadlessAuthentication, error) {
+func (h *HeadlessAuthenticationWatcher) Wait(ctx context.Context, name string, cond WaitConditionFunc) (*types.HeadlessAuthentication, error) {
+	var initialBackendCheck = true
```

Comment:
nit: using `var` for definition is a bit uncommon, use the `:=` shorthand?

```suggestion
	initialBackendCheck := true
```